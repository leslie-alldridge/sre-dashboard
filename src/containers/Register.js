import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import axios from "axios";

import {
  LinkButtons,
  SubmitButtons,
  registerButton,
  homeButton,
  loginButton,
  inputStyle,
  HeaderBar
} from "../components";

const title = {
  pageTitle: "Register Screen"
};

class Register extends Component {
  constructor(props) {
    super(props);

    this.state = {
      first_name: "",
      last_name: "",
      email: "",
      username: "",
      password: "",
      messageFromServer: "",
      showError: false,
      registerError: false,
      loginError: false
    };
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  registerUser = e => {
    e.preventDefault();
    if (
      this.state.username === "" ||
      this.state.password === "" ||
      this.state.email === ""
    ) {
      this.setState({
        showError: true,
        loginError: false,
        registerError: true
      });
    } else {
      axios
        .post("/register", {
          first_name: this.state.first_name,
          last_name: this.state.last_name,
          email: this.state.email,
          username: this.state.username,
          password: this.state.password
        })
        .then(response => {
          console.log(response.data);
          if (response.data === "username or email already taken") {
            this.setState({
              showError: true,
              loginError: true,
              registerError: false
            });
          } else {
            console.log(response.data)
            this.setState({
              //changed here
              messageFromServer: response.data.message,
              showError: false,
              loginError: false,
              registerError: false
            });
          }
        })
        .catch(error => {
          console.log(error.data);
        });
    }
  };

  render() {
    const {
      first_name,
      last_name,
      email,
      username,
      password,
      messageFromServer,
      showError,
      loginError,
      registerError
    } = this.state;

    if (messageFromServer === "") {
      return (
        <div>
          <HeaderBar title={title} />
          <form className="profile-form" onSubmit={this.registerUser}>
            <TextField
              style={inputStyle}
              id="first_name"
              label="first_name"
              value={first_name}
              onChange={this.handleChange("first_name")}
              placeholder="First Name"
            />
            <TextField
              style={inputStyle}
              id="last_name"
              label="last_name"
              value={last_name}
              onChange={this.handleChange("last_name")}
              placeholder="Last Name"
            />
            <TextField
              style={inputStyle}
              id="email"
              label="email"
              value={email}
              onChange={this.handleChange("email")}
              placeholder="Email"
            />
            <TextField
              style={inputStyle}
              id="username"
              label="username"
              value={username}
              onChange={this.handleChange("username")}
              placeholder="Username"
            />
            <TextField
              style={inputStyle}
              id="password"
              label="password"
              value={password}
              onChange={this.handleChange("password")}
              placeholder="Password"
              type="password"
            />
            <SubmitButtons
              buttonStyle={registerButton}
              buttonText={"Register"}
            />
          </form>
          {showError === true && registerError === true && (
            <div>
              <p>Username, password and email are required fields.</p>
            </div>
          )}
          {showError === true && loginError === true && (
            <div>
              <p>
                That username or email is already taken. Please choose another
                or login.
              </p>
              <LinkButtons
                buttonText={`Login`}
                buttonStyle={loginButton}
                link={"/login"}
              />
            </div>
          )}
          <LinkButtons
            buttonText={`Go Home`}
            buttonStyle={homeButton}
            link={"/"}
          />
        </div>
      );
    } else if (messageFromServer === "success") {
      return (
        <div>
          <HeaderBar title={title} />
          <h3>User successfully registered!</h3>
          <LinkButtons
            buttonText={`Go Login`}
            buttonStyle={loginButton}
            link={`/login`}
          />
        </div>
      );
    }
  }
}

export default Register;
