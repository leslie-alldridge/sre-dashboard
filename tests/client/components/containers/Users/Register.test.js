import React from "react";
import { shallow } from "enzyme";
import Register from "../../../../../src/containers/Users/Register";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Register debug />);

    expect(component).toMatchSnapshot();
  });
});
