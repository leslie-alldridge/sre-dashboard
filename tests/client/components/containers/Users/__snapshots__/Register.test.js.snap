// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "Register Screen",
            }
          }
        />,
        <form
          className="profile-form"
          onSubmit={[Function]}
        >
          <TextField
            id="first_name"
            label="first_name"
            onChange={[Function]}
            placeholder="First Name"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            value=""
          />
          <TextField
            id="last_name"
            label="last_name"
            onChange={[Function]}
            placeholder="Last Name"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            value=""
          />
          <TextField
            id="email"
            label="email"
            onChange={[Function]}
            placeholder="Email"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            value=""
          />
          <TextField
            id="username"
            label="username"
            onChange={[Function]}
            placeholder="Username"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            value=""
          />
          <TextField
            id="password"
            label="password"
            onChange={[Function]}
            placeholder="Password"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            type="password"
            value=""
          />
          <SubmitButtons
            buttonStyle={
              Object {
                "background": "green",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Register"
          />
        </form>,
        false,
        false,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "mediumpurple",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Go Home"
          link="/"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "Register Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextField
              id="first_name"
              label="first_name"
              onChange={[Function]}
              placeholder="First Name"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />,
            <TextField
              id="last_name"
              label="last_name"
              onChange={[Function]}
              placeholder="Last Name"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />,
            <TextField
              id="email"
              label="email"
              onChange={[Function]}
              placeholder="Email"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />,
            <TextField
              id="username"
              label="username"
              onChange={[Function]}
              placeholder="Username"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />,
            <TextField
              id="password"
              label="password"
              onChange={[Function]}
              placeholder="Password"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              type="password"
              value=""
            />,
            <SubmitButtons
              buttonStyle={
                Object {
                  "background": "green",
                  "margin": "1em",
                  "padding": "1em",
                }
              }
              buttonText="Register"
            />,
          ],
          "className": "profile-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "first_name",
              "label": "first_name",
              "onChange": [Function],
              "placeholder": "First Name",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "last_name",
              "label": "last_name",
              "onChange": [Function],
              "placeholder": "Last Name",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "email",
              "label": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "username",
              "label": "username",
              "onChange": [Function],
              "placeholder": "Username",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "password",
              "label": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buttonStyle": Object {
                "background": "green",
                "margin": "1em",
                "padding": "1em",
              },
              "buttonText": "Register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "mediumpurple",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Go Home",
          "link": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "Register Screen",
              }
            }
          />,
          <form
            className="profile-form"
            onSubmit={[Function]}
          >
            <TextField
              id="first_name"
              label="first_name"
              onChange={[Function]}
              placeholder="First Name"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />
            <TextField
              id="last_name"
              label="last_name"
              onChange={[Function]}
              placeholder="Last Name"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />
            <TextField
              id="email"
              label="email"
              onChange={[Function]}
              placeholder="Email"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />
            <TextField
              id="username"
              label="username"
              onChange={[Function]}
              placeholder="Username"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />
            <TextField
              id="password"
              label="password"
              onChange={[Function]}
              placeholder="Password"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              type="password"
              value=""
            />
            <SubmitButtons
              buttonStyle={
                Object {
                  "background": "green",
                  "margin": "1em",
                  "padding": "1em",
                }
              }
              buttonText="Register"
            />
          </form>,
          false,
          false,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "mediumpurple",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Go Home"
            link="/"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "Register Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextField
                id="first_name"
                label="first_name"
                onChange={[Function]}
                placeholder="First Name"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                value=""
              />,
              <TextField
                id="last_name"
                label="last_name"
                onChange={[Function]}
                placeholder="Last Name"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                value=""
              />,
              <TextField
                id="email"
                label="email"
                onChange={[Function]}
                placeholder="Email"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                value=""
              />,
              <TextField
                id="username"
                label="username"
                onChange={[Function]}
                placeholder="Username"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                value=""
              />,
              <TextField
                id="password"
                label="password"
                onChange={[Function]}
                placeholder="Password"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                type="password"
                value=""
              />,
              <SubmitButtons
                buttonStyle={
                  Object {
                    "background": "green",
                    "margin": "1em",
                    "padding": "1em",
                  }
                }
                buttonText="Register"
              />,
            ],
            "className": "profile-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "first_name",
                "label": "first_name",
                "onChange": [Function],
                "placeholder": "First Name",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "last_name",
                "label": "last_name",
                "onChange": [Function],
                "placeholder": "Last Name",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "email",
                "label": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "username",
                "label": "username",
                "onChange": [Function],
                "placeholder": "Username",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "password",
                "label": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "buttonStyle": Object {
                  "background": "green",
                  "margin": "1em",
                  "padding": "1em",
                },
                "buttonText": "Register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "mediumpurple",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Go Home",
            "link": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
